cmd/apiserver/server_test.go

```diff
@@ -0,0 +1,462 @@
+			t.Fatal(diff)
+		}
+	}
+}
+
+func TestGetCalculationByName(t *testing.T) {
+	testCases := []struct {
+		id                 string
+		initialCalculation calculationsv1.Calculation
+		expected           calculationsv1.Calculation
+		errorExpected      bool
+	}{
+		{
+			id: "no calculations",
+		},
+		{
+			id: "one calculation returns",
+			initialCalculation: calculationsv1.Calculation{
+				TypeMeta:   metav1.TypeMeta{Kind: "Calculation", APIVersion: "vega.io/v1"},
+				ObjectMeta: metav1.ObjectMeta{Name: fmt.Sprintf("calc-%s", util.InputHash([]byte(fmt.Sprintf("%f", 12100.0)), []byte(fmt.Sprintf("%f", 4.0))))},
+				Phase:      calculationsv1.CreatedPhase,
+				Status:     calculationsv1.CalculationStatus{StartTime: metav1.Time{Time: time.Date(2000, 0, 0, 0, 0, 0, 0, time.UTC)}},
+				Spec:       calculationsv1.CalculationSpec{Teff: 12100.0, LogG: 4.0},
+			},
+		},
+		{
+			id: "get calculation with wrong name",
+			initialCalculation: calculationsv1.Calculation{
+				TypeMeta:   metav1.TypeMeta{Kind: "Calculation", APIVersion: "vega.io/v1"},
+				ObjectMeta: metav1.ObjectMeta{Name: "calc-wrong-name"},
+				Phase:      calculationsv1.CreatedPhase,
+				Status:     calculationsv1.CalculationStatus{StartTime: metav1.Time{Time: time.Date(2000, 0, 0, 0, 0, 0, 0, time.UTC)}},
+				Spec:       calculationsv1.CalculationSpec{Teff: 12100.0, LogG: 4.0},
+			},
+			errorExpected: true,
+		},
+	}
+
+	for _, tc := range testCases {
+		fakecs := fake.NewSimpleClientset()
+		fakeClient := fakecs.VegaV1()
+
+		s := server{
+			logger: logrus.WithField("test-name", tc.id),
+			ctx:    context.Background(),
+			client: fakeClient,
+		}
+
+		_, err := fakeClient.Calculations().Create(s.ctx, &tc.initialCalculation, metav1.CreateOptions{})
+		if err != nil {
+			t.Errorf("Coudn't create the calculation: %v", tc.initialCalculation)
+		} else {
+			logrus.WithFields(logrus.Fields{"name": tc.initialCalculation.Name, "teff": tc.initialCalculation.Spec.Teff, "logG": tc.initialCalculation.Spec.LogG}).Info("Calculation successfully created")
+		}
+		tc.expected = tc.initialCalculation
+
+		req, err := http.NewRequest("GET", "/calculation/", nil)
+		if err != nil {
+			t.Fatal(err)
+		}
+
+		vars := map[string]string{
+			"id": tc.initialCalculation.Name,
+		}
+		req = mux.SetURLVars(req, vars)
+
+		req.Header.Set("Content-Type", "application/json")
+		rr := httptest.NewRecorder()
+		handler := http.HandlerFunc(s.getCalculationByName)
+		handler.ServeHTTP(rr, req)
+
+		var actualData *calculationsv1.Calculation
+
+		err = json.Unmarshal(rr.Body.Bytes(), &actualData)
+		if err != nil {
+			t.Fatal(err)
+		}
+
+		if diff := cmp.Diff(tc.expected, *actualData); diff != "" && !tc.errorExpected {
+			t.Fatal(diff)
+		}
+	}
+}
+
+func TestGetCalculation(t *testing.T) {
+	testCases := []struct {
+		id                 string
+		initialCalculation calculationsv1.Calculation
+		expected           calculationsv1.Calculation
```

Comment:
this should be a list as well.

Reply:
Why should expected be list of calculations? I can return only one calculation as the name says `getCalculation`.

Reply:
Yes, but the cluster holds many calculations and with `getCalculation` we get only one.

Reply:
Oh I get your point. Sorry my bad. You should only have a list of calculation in `initialCalculations`

Reply:
Also is slice of calculations enough or you mean `calculationList`?